src/main/java/htsjdk/tribble/index/tabix/TabixIndex.java

```diff
@@ -219,8 +219,8 @@ public void write(final Path tabixPath) throws IOException {
      */
     @Override
     public void writeBasedOnFeaturePath(final Path featurePath) throws IOException {
-        if (Files.isRegularFile(featurePath)) return;
-        write(IOUtil.getPath(Tribble.tabixIndexPath(featurePath)));
+        if (!Files.isRegularFile(featurePath)) return;
```

Comment:
Ack, good catch.  That's terrible behavior.  I wonder why it was done that way?  Definitely needs to at least log a warning.  I would also prefer your suggestion to fail with a TribbleException, but I think we'd need to investigate if that would break existing downstream code somehow.  For now, could you have it log a warning, and open a new ticket to change the behavior?